clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -660,6 +660,56 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-lm");
   }
 
+  // If requested, use a custom linker script to handle very large device code
+  // sections.
+  if (Args.hasArg(options::OPT_fsycl) &&
+      Args.hasFlag(options::OPT_fsycl_link_huge_device_code,
+                   options::OPT_fno_sycl_link_huge_device_code, false)) {
+    // Create temporary linker script. Keep it if save-temps is enabled.
+    const char *LKS;
+    SmallString<256> Name = llvm::sys::path::filename(Output.getFilename());
+    if (C.getDriver().isSaveTempsEnabled()) {
+      llvm::sys::path::replace_extension(Name, "ld");
```

Comment:
When using `-save-temps` and no output file name, would this end up being named `a.ld`?

Reply:
Yes, this would result in a script named `a.ld`. (Alongside temporary object files like `foo-host-x86_64-unknown-linux-gnu.o` and `a-sycl-spir64-unknown-unknown.o`.)